src/plugins/intel_gpu/src/runtime/ocl/ocl_device_detector.cpp

```diff
@@ -91,47 +92,58 @@ static std::vector<cl::Device> getSubDevices(cl::Device& rootDevice) {
     return subDevices;
 }
 
+std::vector<device::ptr> ocl_device_detector::sort_devices(const std::vector<device::ptr>& devices_list) {
```

Comment:
Why dont we just add some weight for integrated_gpu/INTEL_GPU/others and then sort it using some priority queue or etc?

Reply:
updated

Reply:
If the priority is same, (e.g., multiple INTEL dgpu devices ?) should it be sorted as inserted order?

Reply:
I think it's not really critical, but changed to stable_sort